docs/rfc/rfc-1.adoc

```diff
@@ -312,10 +312,43 @@ to be included in a sweep, we do it. Otherwise, we schedule the next sweep for
 
 Here, we're making the tradeoff between reducing fees (having less frequent
 batches) and increasing reliability from a user experience standpoint. This
-setup allows us to have two active wallets that are always staggered from one
+setup allows us to have two active wallets that are always <<staggering,staggered>> from one
 another, so the worst case scenario for the next sweep is always half of
 `sweep_period`.
 
+[[staggering]]
+===== Active Wallet Sweep Staggering
+Governable Parameters
+
+- `sweep_period`: The amount of time we wait between sweep-period-sweeps.
+
+Since the youngest two wallets both sweep deposits based on their
+`sweep_period`, we can improve our user experience by intentionally staggering
+the start time of these period sweeps to be equally spaced apart from each
+other.
+
+Good Example:
+- `sweep_period = 8 hours`
+- `wallet_1` sweeps at 12:00, 20:00, and 04:00
+- `wallet_2` sweeps at 16:00, 00:00, and 08:00
+
+Bad Example:
+- `sweep_period = 8 hours`
+- `wallet_1` sweeps at 12:00, 20:00, and 04:00
+- `wallet_2` sweeps at 13:00, 21:00, and 05:00
+
+In the "Good Example", there are 6 worst case scenarios: a user deposits at
+12:01, 20:01, 4:01, 16:01, 00:01, and 08:01. Those deposits would be swept at
+16:00, 24:00, 8:00, 20:00, 04:00, and 12:00 respectively, and the amount of
+time to wait is 3h59m.
+
+In the "Bad Example", there are 3 worst case scenarios: a user deposits at
+13:01, 21:01, and 05:01. Those deposits would be swept at 20:00, 04:00, and
+12:00 respectively, and the amount of time to wait is 6h59m.
+
+By intentionally offsetting the sweep schedule, we're able to reduce the worst
+case for a deposit significantly!
```

Comment:
Let's say that `wallet_1` is the latest wallet. It will mean that all new deposits created from the Dashboard will be directed to that wallet, right? So I think in the worst case scenario (deposits created at 12:01, 20:01, 04:01) the user will need to wait almost 8h for the sweep to happen.
Or am I missing something?
If I'm right, then I don't see a gain from having sweeps on `wallet_1` and `wallet_2` executed at different times and think that having them executed the same time would be clearer for users (especially if in the Dashboard we'll be showing the next sweeping time for the primary wallet).

Reply:
This combines a couple of scattered ideas:

- The youngest two wallets are "active" wallets, meaning that they're responsible for sweeping
- The dApp gets to choose which wallet to direct deposits to
- Having more than one active wallet is good for redundancy / DoS protection

The dApp (and non-dApp users) can point to whichever wallet will sweep next rather than the youngest wallet, which should decrease wait times if we do it well

---

also, after typing this out, I'm wondering if having more than two active wallets is worth investigating

Reply:
Oh, I see, I was missing this piece of info:
> The dApp (and non-dApp users) can point to whichever wallet will sweep next rather than the youngest wallet.
With this, it makes sense to do the staggering.

I previously thought that we did the sweeping of the second (older) wallet only to allow for sweeping of the deposits which were assigned to that wallet just before that wallet stopped being the newest one.

I wonder now, as it will be possible to choose in the dapp which of the "active" wallets user wants to deposit to (and user may select a wallet which is soon going to become not-"active"), if we should do something to allow for sweeping of the deposits performed to the wallet that becomes not-"active" before user reveals the deposit. If we don't do anything, user may end up with his funds blocked for 30 days.
For example we could:
- not allow for choosing for a deposit a wallet that is close to become "not-active"
- show a warning next to the problematic wallet informing about potential consequences of choosing it for deposit (or indicate in some other way that the wallet may soon become not-"active").
First option seems safer to me...

Reply:
Yeah - I think how the dApp wants to handle it could end up being complex (and ultimately really cool). I talked it over w/ @Battenfield  during the standup today and he seemed excited about it!

Anyway - is there anything specific you want to see in this PR or in a follow up? Super down to elaborate!

Reply:
It would be good to show somewhere in the doc an information about the possibility to choose one of the active wallets when depositing.
In the Overview we have following sentence, which suggests that users can deposit to only one address:
> Depositors send BTC funds to the most-recently-created-wallet by using pay-to-script-hash (P2SH) which contains hashed information about the depositor’s minting ethereum address."

We don't have to do that in this PR, though. I know that there's a discussion regarding how many active wallets there should be (and if they should be created periodically), so we may do the update once we have a final decision on that.

Reply:
It would be good to show somewhere in the doc an information about the possibility to choose one of the active wallets when depositing.
In the Overview we have following sentence, which suggests that users can deposit to only one address:

Depositors send BTC funds to the most-recently-created-wallet by using pay-to-script-hash (P2SH) which contains hashed information about the depositor’s minting ethereum address."

We don't have to do that in this PR, though. I know that there's a discussion regarding how many active wallets there should be (and if they should be created periodically), so we may do the update once we have a final decision on that.

Reply:
I will capture it in a follow-up issue and merge this PR as-is.